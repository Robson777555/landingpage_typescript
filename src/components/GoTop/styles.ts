import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;