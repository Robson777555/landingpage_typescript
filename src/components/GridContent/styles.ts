// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;