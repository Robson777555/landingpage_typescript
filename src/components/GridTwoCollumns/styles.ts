// src/components/GridTwoCollumns/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      text-align: center;
      justify-content: center;
      gap: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.medium};

    /* Estilos para mobile */
    @media ${theme.media.lteMedium} {
      align-items: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.medium};
    max-width: 400px;
    width: 100%;

    img, .next-image {
      max-width: 100%;
      height: auto;
      width: 100%;
      object-fit: contain;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} 0;
      max-width: 300px;
    }
  `}
`;

export const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
