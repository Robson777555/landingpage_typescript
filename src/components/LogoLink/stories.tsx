import { Meta, StoryObj } from "@storybook/react";
import { LogoLink } from "./LogoLink";

const meta: Meta<typeof LogoLink> = {
  title: "Components/LogoLink",
  component: LogoLink,
  parameters: {
    layout: "centered",
    nextjs: {
      appDirectory: true,
    },
  },
  argTypes: {
    text: {
      control: "text",
      description: "Texto do logo (se n√£o usar imagem)",
    },
    srcImg: {
      control: "text",
      description: "Caminho para a imagem do logo",
    },
    link: {
      control: "text",
      description: "URL de destino do link",
    },
  },
  args: {
    text: "My Logo",
    srcImg: "",
    link: "/",
  },
};

export default meta;

type Story = StoryObj<typeof LogoLink>;

export const TextLogo: Story = {
  args: {
    text: "Text Logo",
    srcImg: "",
    link: "/",
  },
};

export const ImageLogo: Story = {
  args: {
    text: "Image Logo",
    srcImg: "/logo.svg",
    link: "https://example.com",
  },
};

export const Combined: Story = {
  args: {
    text: "Logo + Text",
    srcImg: "/logo.svg",
    link: "/about",
  },
};
