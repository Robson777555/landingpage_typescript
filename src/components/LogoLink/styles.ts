import styled from 'styled-components';

export const Container = styled.a`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spacings.small};
  text-decoration: none;
  color: inherit;
  font-size: ${({ theme }) => theme.fonts.sizes.large};
  font-weight: bold;
  text-transform: uppercase;
  transition: opacity 0.3s ease;

  &:hover {
    opacity: 0.8;
  }

  > img {
    max-height: 3rem;
    width: auto;
    object-fit: contain;
  }
`;