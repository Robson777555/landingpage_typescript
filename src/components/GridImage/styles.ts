// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
