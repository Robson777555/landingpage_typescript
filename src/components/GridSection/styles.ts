import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
