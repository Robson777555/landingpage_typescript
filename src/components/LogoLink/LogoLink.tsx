'use client';
import React from 'react';
import Image from 'next/image';
import Link from 'next/link';
import * as Styled from './styles';
import { Heading } from '../Heading/Heading';

interface LogoLinkProps {
  text: string;
  srcImg?: string;
  link?: string;
}

export const LogoLink = ({ text, srcImg = '', link = '/' }: LogoLinkProps) => {
  return (
    <Heading size="small" uppercase>
      <Styled.Container
        href={link}
        as={Link}
        target={link.startsWith('http') ? '_blank' : '_self'}
        rel={link.startsWith('http') ? 'noopener noreferrer' : undefined}
      >
        {srcImg ? (
          <Image
            src={srcImg}
            alt={text}
            width={48}
            height={48}
            style={{
              height: '3rem',
              width: 'auto',
            }}
            priority
          />
        ) : (
          <span>{text}</span>
        )}
      </Styled.Container>
    </Heading>
  );
};