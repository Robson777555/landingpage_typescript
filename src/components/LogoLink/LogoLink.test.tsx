import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { LogoLink } from './LogoLink';
import { theme } from '../../styles/theme';
import { describe, it, expect } from 'vitest';

describe('<LogoLink />', () => {
  const renderLogoLink = (props = {}) => {
    const defaultProps = {
      text: 'Logo',
      link: '#target',
    };

    return render(
      <ThemeProvider theme={theme}>
        <LogoLink {...defaultProps} {...props} />
      </ThemeProvider>
    );
  };

  it('should render text logo with correct attributes', () => {
    renderLogoLink({ text: 'My Logo' });

    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('href', '#target');
    expect(link).toHaveAttribute('target', '_self');
    expect(link).not.toHaveAttribute('rel');
    expect(screen.getByText('My Logo')).toBeInTheDocument();
  });

  it('should render image logo with alt text', () => {
    renderLogoLink({ srcImg: '/logo.png', text: 'My Logo' });

    const image = screen.getByRole('img', { name: 'My Logo' });
    expect(image).toBeInTheDocument();
  });

  it('should match snapshot with text only', () => {
    const { container } = renderLogoLink();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with image', () => {
    const { container } = renderLogoLink({ srcImg: '/logo.png' });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should add noopener and noreferrer for external links', () => {
    renderLogoLink({ link: 'https://external.com' });
    const link = screen.getByRole('link');
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveAttribute('rel', 'noopener noreferrer');
  });
  it ('Snapshot of page', () => {
    renderLogoLink();
    const link = screen.getByRole('link');
    expect(link).toMatchSnapshot();
  })
});