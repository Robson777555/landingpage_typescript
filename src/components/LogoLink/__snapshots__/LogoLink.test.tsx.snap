// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LogoLink /> > Snapshot of page 1`] = `
<a
  class="sc-bXdtCk joylCA"
  href="#target"
  target="_self"
>
  <span>
    Logo
  </span>
</a>
`;

exports[`<LogoLink /> > should match snapshot with image 1`] = `
<h1
  class="sc-hKDTPf fAiovE"
  data-testid="heading"
>
  <a
    class="sc-bXdtCk joylCA"
    href="#target"
    target="_self"
  >
    <img
      alt="Logo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="48"
      src="/_next/image?url=%2Flogo.png&w=96&q=75"
      srcset="/_next/image?url=%2Flogo.png&w=48&q=75 1x, /_next/image?url=%2Flogo.png&w=96&q=75 2x"
      style="color: transparent; height: 3rem; width: auto;"
      width="48"
    />
  </a>
</h1>
`;

exports[`<LogoLink /> > should match snapshot with text only 1`] = `
<h1
  class="sc-hKDTPf fAiovE"
  data-testid="heading"
>
  <a
    class="sc-bXdtCk joylCA"
    href="#target"
    target="_self"
  >
    <span>
      Logo
    </span>
  </a>
</h1>
`;
